services/core-api/app/api/now_applications/resources/now_application_delay_resource.py

```diff
@@ -0,0 +1,72 @@
+from flask_restplus import Resource
+from flask import request, current_app
+
+from werkzeug.exceptions import BadRequest, NotFound, InternalServerError
+from app.extensions import api
+from app.api.utils.access_decorators import requires_role_view_all
+from app.api.utils.custom_reqparser import CustomReqparser
+from app.api.utils.resources_mixins import UserMixin
+from werkzeug.exceptions import BadRequest, NotFound, InternalServerError
+
+from app.api.now_applications.models.now_application_delay_type import NOWApplicationDelayType
+from app.api.now_applications.models.now_application_delay import NOWApplicationDelay
+from app.api.now_applications.models.now_application_identity import NOWApplicationIdentity
+from app.api.now_applications.response_models import NOW_APPLICATION_DELAY_TYPE, NOW_APPLICATION_DELAY
+
+
+class NOWApplicationDelayTypeResource(Resource, UserMixin):
+    @api.doc(description='Get a list of all Notice of Work Delay Reasons.', params={})
+    @requires_role_view_all
+    @api.marshal_with(NOW_APPLICATION_DELAY_TYPE, code=200, envelope='records')
+    def get(self):
+        return NOWApplicationDelayType.get_all()
+
+
+class NOWApplicationDelayListResource(Resource, UserMixin):
+    @api.doc(description='Get a list of all Notice of Work Delay Reasons.', params={})
+    @requires_role_view_all
+    @api.marshal_with(NOW_APPLICATION_DELAY, code=200, envelope='records')
+    def get(self, now_application_guid):
+        return NOWApplicationIdentity.find_by_guid(now_application_guid).application_delays
```

Comment:
Need to check for null here and return `NotFound`, otherwise, there's potential for uncaught null reference exception.

Reply:
good eye, and added test.